Design.md

```diff
@@ -0,0 +1,128 @@
+# Job Worker Design Doc
+
+Design doc for a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Level 2 requirements
+
+- Library: Worker library with methods to start/stop/query status and get the output of a job.
+- API:
+    - HTTPS API to start/stop/get status of a running process.
+    - Use mTLS authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto
+      setup for certificates. Do not use any other authentication protocols on top of mTLS.
+    - Use a simple authorization scheme.
+- CLI:
+    - CLI should be able to connect to worker service and schedule several jobs.
+    - CLI should be able to query result of the job execution and fetch its output.
+
+## Job object
+
+- `ID` unique identifier of a job
+- `Command` linux command to be executed
+- `Status` current status of the job
+- `Output` output of the job process
+- `Error` exit error or, termination error
+- `Timeout` timeout in seconds to stop the job
+
+## Job status
+
+- `Created` Received a job creation request (short-lived since we immediately start the job instead of queueing)
+- `Running` Job process started
+- `Running_Timeout` Failed to kill process after timeout
+- `Running_Force_Stop` Failed to kill process after force stop requested
+- `Exited` Job finished successfully
+- `Exited_Error` Job finished with error
+- `Exited_Timeout` Job exited after timeout
+- `Exited_Force_Stop` Job exited after force stop requested
+
+## Job handling
+
+- Wrap the linux command in `bash`
+- Start the process using `exec.Cmd` in separate go routine
+- Utilize go channels to communicate with the go routine
+- The go routine should exit when job finishes or timeout reached or force stop requested
+
+## Library
+
+- `CreateJob(command, timeout)`
+    - Generate a unique ID
```

Comment:
how will the ID be generated?

Reply:
Using the [rs/xid](github.com/rs/xid) library we can generate a globally unique ID.